designer/server/src/main/scala/pl/touk/nussknacker/ui/process/processingtype/ScenarioParametersDeterminer.scala

```diff
@@ -24,15 +25,16 @@ object ScenarioParametersDeterminer {
       case Nil => throw new IllegalStateException(s"Empty list of components for processing type: $processingType")
       case nonEmptyList =>
         val intersection = nonEmptyList.foldLeft(ProcessingMode.all) {
-          case (acc, Some(componentProcessingModes)) => acc.intersect(componentProcessingModes)
-          case (acc, None)                           => acc
+          case (acc, AllowedProcessingModes.SetOf(componentProcessingModes)) =>
+            acc.intersect(componentProcessingModes.toSortedSet)
+          case (acc, AllowedProcessingModes.AllProcessingModes) => acc
         }
         intersection.toList match {
           case oneMode :: Nil => oneMode
           // FIXME: some proper errors and tests for that
           case Nil =>
             val componentsWithDefinedAllowedProcessingModes = componentsToProcessingMode.collect {
```

Comment:
WDYT about changing `collect` to `flatMap` here? IMO `collect` is not so safe in the context of potential extending sealed trait